### PR TITLE
make-move-migration

### LINKED ISSUE #387: Migrate from fetch API to Axios
`src/services/game-apis.ts` uses fetch api currently, we need to migrate to axios for built‑in JSON parsing, better error handling, and timeout support

### PR DESCRIPTION
**Pull request by Rakshitg600**
## Description
migrate make move api

## Category (replace the empty space inside brackets with x for tick)
- [ ] Bug
- [x] Refactor
- [ ] Enhancement/New Feature
- [ ] Tests
- [ ] Documentation

## Summary by CodeRabbit

* **Refactor**
  * Single‑player flow streamlined (reset → quit); client init now gated by auth readiness and unified startup/reset handling.
  * Core game engine and server-side session handling moved/removed, shifting responsibility away from in-app session persistence.

* **New Features**
  * Wallet integration: fetches and updates coins and XP on sign‑in and after game actions.
  * Client API calls standardized with axios and stricter response schemas.

* **Bug Fixes**
  * More consistent handling of move/undo/skip outcomes and init edge cases.

* **Tests**
  * Mocks updated to include wallet, coins, and XP.

* **Chores**
  * Removed realtime listener, local session store export, server API routes, AI/reward utilities, DB/Firestore helpers, test setup hooks, and pre‑commit tooling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/app/vsComputer/page.tsx [hunk 1]
```diff
@@ -217,21 +213,42 @@ const Game = () => {
...
 					}
-				} else if ("error" in data) {
-					toast.error(data.error || "Invalid move");
+					setXP(XP + resp.xpRewarded);
+					setActiveModal("winner");
+					playWinSound(sfxMute);
 				} else {
-					toast.error("Unexpected response from server");
+					playMoveSound(sfxMute);
 				}
```
[the remark this answers is not in the document]
Comment (Rakshitg600) [reply src/app/vsComputer/page.tsx:245]: I am choosing consistency over availability here. considering server as the source of truth, the frontend shows stale data unless api call to get-wallet is completed. If required we can switch to availability over consistency later.

File: src/app/vsComputer/page.tsx [hunk 2]
```diff
@@ -249,24 +266,19 @@ const Game = () => {
...
-				}
-			} else {
+			if (!user) {
 				toast.error("User not authenticated");
 				router.push("/");
+				return;
 			}
+
+			const data = await quitGame(sessionId, await user.getIdToken());
+			console.log(data);
```
[the remark this answers is not in the document]
Comment (Rakshitg600) [reply src/app/vsComputer/page.tsx:269]: the production mode in next.config.ts ensures that no logs are shown in production and still allowing the dev mode to see these logs which can help developers while working. All the logs can be cleaned and replaced with good error and exception handling later in a dedicated PR.

File: src/app/vsComputer/page.tsx [hunk 3]
```diff
@@ -347,33 +388,21 @@ const Game = () => {
...
```
[the remark this answers is not in the document]
Comment (Rakshitg600) [reply src/app/vsComputer/page.tsx:412]: the production mode in next.config.ts ensures that no logs are shown in production and still allowing the dev mode to see these logs which can help developers while working. All the logs can be cleaned and replaced with good error and exception handling later in a dedicated PR.
